backend/src/controllers/playlists.ts

```diff
@@ -1,26 +1,27 @@
 import { Request, Response } from "express";
 import { SpotifyService } from "../services/spotify";
+import type { Playlists, Tracks } from "@types"
 
 export class PlaylistsController {
     constructor(private spotifyService: SpotifyService) {}
 
     async getPlaylists(req: Request, res: Response) {
         const accessToken = req.cookies.accesstoken;
-        if (!accessToken) {
+        if (!accessToken || typeof accessToken !== "string")  {
             res.status(401).json({ error: "Unauthorized" });
             return;
         }
-        const playlists = await this.spotifyService.getPlaylists(accessToken);
-        res.json(playlists);
+        const playlists: Playlists = await this.spotifyService.getPlaylists(accessToken);
+        res.json({ playlists });
     }
 
     async getTracks(req: Request, res: Response) {
-        const accessToken = req.cookies.accesstoken;
-        if (!accessToken) {
+        const accessToken = req.cookies?.accesstoken;
+        if (!accessToken || typeof accessToken !== "string")  {
             res.status(401).json({ error: "Unauthorized" });
             return;
         }
-        const tracks = await this.spotifyService.getTracks(accessToken, req.params.playlistId);
-        res.json(tracks);
+        const tracks: Tracks = await this.spotifyService.getTracks(accessToken, req.params.playlistId);
+        res.json({ tracks });
```

Comment:
I recommend making the controller formats more consistent. Do you prefer something like `res.json(tracks)` or  `res.json({ tracks })`? 

I prefer the later because it's easier to expand the response later, like error messages or meta data.